SeafFileProvider/SeafEnumerator.m

```diff
@@ -86,12 +86,12 @@ - (void)enumerateItemsForObserver:(id<NSFileProviderEnumerationObserver>)observe
             [dir loadContentSuccess: ^(SeafDir *d) {
                 NSArray *items = [self getSeafDirProviderItems:d startingAtPage:page];
                 [observer didEnumerateItems:items];
-                if (items.count == self.maxItemCount) {
+                if (items.count != self.maxItemCount) {
```

Comment:
Use items.count != self.maxItemCount if sufficient to determine if it reaches the end, but it is not a requirement for the last chunk.
So if the last chunk has exactly self.maxItemCount elements, it will still require the next page, which will call getSeafDirProviderItems again, but this time it will return an empty array. 
It works, but just a little weird. Could you please make the code logic clearer.

You can directly calculate how many items have been requested, then you will know if it reaches the end.